include/PathOrderOptimizer.h

```diff
@@ -759,81 +740,124 @@ class PathOrderOptimizer
         constexpr double weight_consistency_x = 0.05;
         constexpr double weight_consistency_y = weight_consistency_x / 2.0; // Less weight on Y to avoid symmetry effects
 
-        // Fixed divider for shortest distances computation. The divider should be set so that the minimum encountered
-        // distance gives a score very close to 1.0, and a medium-far distance gives a score close to 0.5
-        constexpr double distance_divider = 20.0;
+        // ########## Step 2: define which criteria should be taken into account in the total score
+        const bool calculate_forced_pos_score = path.force_start_index_.has_value();
+
+        // For most seam types, the shortest distance matters. Not for SHARPEST_CORNER though.
+        const bool calculate_distance_score
+            = ! calculate_forced_pos_score
+           && (path.seam_config_.type_ != EZSeamType::SHARPEST_CORNER || path.seam_config_.corner_pref_ == EZSeamCornerPrefType::Z_SEAM_CORNER_PREF_NONE);
+
+        const bool calculate_corner_score
+            = ! calculate_forced_pos_score
+           && (path.seam_config_.type_ == EZSeamType::SHARPEST_CORNER
+               && (path.seam_config_.corner_pref_ != EZSeamCornerPrefType::Z_SEAM_CORNER_PREF_NONE && path.seam_config_.corner_pref_ != EZSeamCornerPrefType::PLUGIN));
+
+        const bool calculate_random_score = ! calculate_forced_pos_score && (path.seam_config_.type_ == EZSeamType::RANDOM);
```

Comment:
I need to do a double (triple) read on the `!flag && condition` pattern. I find this simpler to read;
```suggestion
        bool calculate_distance_score = false;
        bool calculate_corner_score = false;
        bool calculate_random_score = false;
        if (! calculate_forced_pos_score)
        {
            calculate_distance_score = path.seam_config_.type_ != EZSeamType::SHARPEST_CORNER || path.seam_config_.corner_pref_ == EZSeamCornerPrefType::Z_SEAM_CORNER_PREF_NONE;

             calculate_corner_score
            = path.seam_config_.type_ == EZSeamType::SHARPEST_CORNER
               && (path.seam_config_.corner_pref_ != EZSeamCornerPrefType::Z_SEAM_CORNER_PREF_NONE && path.seam_config_.corner_pref_ != EZSeamCornerPrefType::PLUGIN);

             calculate_random_score = path.seam_config_.type_ == EZSeamType::RANDOM;
        }
```

Reply:
It is indeed, but then you loose the constness on the bools, so I'm shared

Reply:
I wish I could say in C++ "from now on, this variable is const" that is a use-case I regularly come up with